backend/src/services/pki-sync/pki-sync-service.ts

```diff
@@ -245,75 +302,78 @@ export const pkiSyncServiceFactory = ({
       projectId
     });
 
-    ForbiddenError.from(permission).throwUnlessCan(
-      ProjectPermissionPkiSyncActions.Read,
-      subject(ProjectPermissionSub.PkiSyncs, { projectId })
-    );
+    ForbiddenError.from(permission).throwUnlessCan(ProjectPermissionPkiSyncActions.Read, ProjectPermissionSub.PkiSyncs);
 
     const pkiSyncs = await pkiSyncDAL.findByProjectId(projectId);
-    return pkiSyncs;
-  };
-
-  const listPkiSyncsBySubscriberId = async ({ subscriberId }: TListPkiSyncsBySubscriberId) => {
-    const pkiSyncs = await pkiSyncDAL.findBySubscriberId(subscriberId);
-    return pkiSyncs;
+    return pkiSyncs as TPkiSync[];
   };
 
   const findPkiSyncById = async ({ id, projectId }: TFindPkiSyncByIdDTO, actor: OrgServiceActor) => {
+    const pkiSync = await pkiSyncDAL.findById(id);
+    if (!pkiSync)
+      throw new NotFoundError({
+        message: `Could not find PKI Sync with ID "${id}"`
+      });
+
+    if (projectId && pkiSync.projectId !== projectId) {
+      throw new NotFoundError({
+        message: `Could not find PKI Sync with ID "${id}" in project "${projectId}"`
+      });
+    }
+
     const { permission } = await permissionService.getProjectPermission({
       actor: actor.type,
       actorId: actor.id,
       actorAuthMethod: actor.authMethod,
       actorOrgId: actor.orgId,
       actionProjectType: ActionProjectType.CertificateManager,
-      projectId
+      projectId: pkiSync.projectId
     });
 
-    const pkiSync = await pkiSyncDAL.findByIdAndProjectId(id, projectId);
-    if (!pkiSync)
-      throw new NotFoundError({
-        message: `Could not find PKI Sync with ID "${id}"`
-      });
+    let findSubscriber;
+    if (pkiSync.subscriberId) {
+      findSubscriber = await pkiSubscriberDAL.findById(pkiSync.subscriberId);
+    }
 
     ForbiddenError.from(permission).throwUnlessCan(
       ProjectPermissionPkiSyncActions.Read,
-      subject(ProjectPermissionSub.PkiSyncs, {
-        projectId,
-        subscriberId: pkiSync.subscriberId
-      })
+      findSubscriber
+        ? subject(ProjectPermissionSub.PkiSyncs, { subscriberName: findSubscriber.name })
+        : ProjectPermissionSub.PkiSyncs
```

Comment:
IGNORE PLACEMENT OF COMMENT

I don't see the subscriber name conditions.. instead I'm seeing secret-related conditions
<img width="1498" height="779" alt="image" src="https://github.com/user-attachments/assets/06d8eeab-47f4-4c1d-942f-69301c2fc552" />